classes/mix-rustler.bbclass

```diff
@@ -0,0 +1,34 @@
+inherit mix-rebar3 cargo_common
+
+# We don't want to use bitbakes vendoring
+CARGO_DISABLE_BITBAKE_VENDORING = "1"
+
+# the binary we will use
+CARGO = "cargo"
+
+# We need cargo to compile for the target
+BASEDEPENDS:append = " cargo-native"
+
+# Ensure we get the right rust variant
+DEPENDS:append:class-target = " rust-native ${RUSTLIB_DEP}"
+DEPENDS:append:class-nativesdk = " rust-native ${RUSTLIB_DEP}"
```

Comment:
This does not work for me (kirkstone for armv7 platform). I get a compilation error:

```
$ bitbake hello-elixir-rustler

...
    Compiling regex-syntax v0.6.27
|    Compiling syn v1.0.100
|    Compiling void v1.0.2
|    Compiling heck v0.4.0
|     Building [                            ] 0/28: memchr(build.rs), regex-s...
| warning: target json file contains unused fields: has-elf-tls
| 
| error[E0461]: couldn't find crate `core` with expected target triple armv7-unknown-linux-gnueabihf
|   |
|   = note: the following crate versions were found:
|           crate `core`, target triple arm-fslc-linux-gnueabi: /opt/yocto/build_lbel/tmp/work/cortexa7t2hf-neon-fslc-linux-gnueabi/hello-elixir-rustler/0.1.0-gitAUTOINC+ff7ba3d7fe-r0/recipe-sysroot/usr/lib/rust/libcore-728ab5d900883ea9.rmeta
|           crate `core`, target triple arm-fslc-linux-gnueabi: /opt/yocto/build_lbel/tmp/work/cortexa7t2hf-neon-fslc-linux-gnueabi/hello-elixir-rustler/0.1.0-gitAUTOINC+ff7ba3d7fe-r0/recipe-sysroot/usr/lib/rust/libcore-728ab5d900883ea9.rlib
| 
| error[E0461]: couldn't find crate `compiler_builtins` with expected target triple armv7-unknown-linux-gnueabihf
|   |
...
```

Reply:
Indeed, that will not work for kirkstone. This change only makes sense for master (langdale).

Reply:
If you are using kirkstone, it is a good point to stick with kirstone branch (YP/OE, meta-erlang and other layers as well).

Reply:
Yes, i am only using the kirkstone branches for everything, but i wanted to test this here, because `LAYERSERIES_COMPAT_erlang-layer` in master still contains kirkstone.

Should kirkstone be removed there or maybe a small comment added before this lines in classes/mix-rustler.bbclass to inform the user that this does not work for crosscompiling when using kirkstone?

Reply:
hmm good point. I'll change it to follow YP/OE release way. Usually they drop others releases.

BTW, in meta-erlang master branch it might be necessary some changes to fix the issue https://github.com/meta-erlang/meta-erlang/issues/205 that might lead to some incompatibility between master and kirskstone branches, not sure yet.